src/main/kotlin/org/rust/ide/template/postfix/stringBasedPostfixTemplates.kt

```diff
@@ -0,0 +1,136 @@
+package org.rust.ide.template.postfix
+
+import com.intellij.codeInsight.template.Template
+import com.intellij.codeInsight.template.impl.TextExpression
+import com.intellij.codeInsight.template.postfix.templates.StringBasedPostfixTemplate
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.RustBinaryExprElement
+import org.rust.lang.core.psi.RustExprElement
+import org.rust.lang.core.psi.RustTupleFieldDeclElement
+import org.rust.lang.core.psi.impl.RustPsiImplUtil
+import org.rust.lang.core.psi.util.descendentsOfType
+import org.rust.lang.core.psi.util.parentOfType
+import org.rust.lang.core.resolve.RustResolveEngine
+import org.rust.lang.core.resolve.scope.RustResolveScope
+import org.rust.lang.core.types.RustEnumType
+import org.rust.lang.core.types.util.resolvedType
+
+class AssertPostfixTemplate : StringBasedPostfixTemplate(
+    "assert",
+    "assert!(expr);",
+    RustTopMostInScopeSelector(RustExprElement::isBool)
+) {
+    override fun getTemplateString(element: PsiElement): String? =
+        "assert!(\$expr$);\$END$"
+
+    override fun getElementToRemove(expr: PsiElement): PsiElement = expr
+}
+
+class AssertEqPostfixTemplate : StringBasedPostfixTemplate(
```

Comment:
Can we make a single intelligent template for both `Assert` and `AssertEq`?

`foo && bar.assert` => `assert!(foo && bar)`

`foo == bar.assert` => `assert_eq!(foo, bar)`

?